src/Surfnet/ServiceProviderDashboard/Domain/Entity/Entity/Coin.php

```diff
@@ -121,4 +112,32 @@ public function getOidcClient()
     {
         return $this->oidcClient;
     }
+
+    /**
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     * @SuppressWarnings(PHPMD.NPathComplexity)
+     */
+    public function merge(?Coin $coin)
```

Comment:
Making null objects for the value objects of the ManageEntity did cross my mind. But was hesitant to make them in the end. The added benefit seemed small, and the raised complexity during construction time made me decide not to make them.

Or can you change my mind?

Reply:
Then I would suggest to at least make the vo argument required and move the check to the caller? That way the methods only responsibility is to merge an object. Currently it seems that for example in Metadata the VO's could also be `null` so calling `merge` on them would then fail. This could happen for example when a logourl is an empty string. So if a field could be made required please do so because that will greatly reduce bugs in the future.

Reply:
Yes, that is a good trade-off for now. Venturing into null objects is much easier that way down the road.